src/main/java/scripts/uploadscripts/SummariesToBigQuery.java

```diff
@@ -0,0 +1,231 @@
+   * Result Uploader based on the current Upload List, and can be used by the upload script methods.
+   *
+   * @param parameters list of string parameters supplied by the upload list
+   */
+  @Override
+  public void setParameters(List<String> parameters) throws Exception {
+    if (parameters == null || parameters.size() < 2) {
+      throw new IllegalArgumentException(
+          "Must provide BigQuery project_id and dataset_name in the parameters list");
+    }
+    projectId = parameters.get(0);
+    datasetName = parameters.get(1);
+  }
+
+  private static String testRunTableName = "testRun";
+  private static String testScriptResultsTableName = "testScriptResults";
+  private static String measurementCollectionTableName = "measurementCollection";
+
+  /**
+   * Upload the test results saved to the given directory. Results may include Test Runner
+   * client-side output and any relevant measurements collected.
+   */
+  @Override
+  public void uploadResults(
+      Path outputDirectory, ServiceAccountSpecification uploaderServiceAccount) throws Exception {
+    // get a BigQuery client object
+    logger.debug("BigQuery project_id:dataset_name: {}:{}", projectId, datasetName);
+    BigQuery bigQueryClient =
+        BigQueryUtils.getClientForServiceAccount(uploaderServiceAccount, projectId);
+    // read in TestConfiguration TestRunSummary, and array of
+    // MeasurementCollectionScript.MeasurementResultSummary objects
+    renderedTestConfiguration = TestRunner.getRenderedTestConfiguration(outputDirectory);
+    testRunSummary = TestRunner.getTestRunSummary(outputDirectory);
+    measurementCollectionSummaries =
+        MeasurementCollector.getMeasurementCollectionSummaries(outputDirectory);
+
+    // insert a single row into testRun
+    if (BigQueryUtils.checkRowExists(
+        bigQueryClient, projectId, datasetName, testRunTableName, "id", testRunSummary.id)) {
+      logger.warn(
+          "A row with this id already exists in the "
+              + testRunTableName
+              + " table. Inserting a duplicate.");
+    }
+    TableId tableId = TableId.of(datasetName, testRunTableName);
+    InsertAllRequest insertRequest =
+        InsertAllRequest.newBuilder(tableId).addRow(buildTestRunRow(outputDirectory)).build();
+    BigQueryUtils.insertAllIntoBigQuery(bigQueryClient, insertRequest);
+
+    // insert into testScriptResults
+    tableId = TableId.of(datasetName, testScriptResultsTableName);
+    InsertAllRequest.Builder insertRequestBuilder = InsertAllRequest.newBuilder(tableId);
+    // Store test summaries in a Map with test names as keys
+    Map<String, TestScriptResult.TestScriptResultSummary> testScriptResultSummaries =
+        new ConcurrentHashMap<String, TestScriptResult.TestScriptResultSummary>();
+    testRunSummary.testScriptResultSummaries.stream()
+        .forEach(
+            runSummary -> testScriptResultSummaries.put(runSummary.testScriptName, runSummary));
```

Comment:
This is the fix for the upload failure in WSM. Previously, this was using the `testScriptDescription` as the key and then using the `testScriptSpecification.name` below to look things up. ([ref](https://github.com/DataBiosphere/terra-workspace-manager/blob/a757d244ae0c4b7b37fcdddccab883807a3fb373/integration/src/main/java/scripts/uploadscripts/SummariesToBigQuery.java#L101) to previous code, which lives in WSM).

Reply:
Thanks for fixing this, totally makes sense!